### DISCUSSION TITLE
log in on a graphical shell and xrdp at the same time

### DISCUSSION THREAD

**Discussion by pharaonic-faery** [neutrinolabs/xrdp#2729] [Q&A] [answered]
Hello!
I'm running xrdp on Raspberry Pi OS and Debian, and I would like to have the possibility to connect to an Xfce local session *and* a remote one through xrdp.
If an Xfce session has been opened locally, I can't connect to a new one though xrdp. If I connect to my Xfce session through xrdp, I can't open an Xfce local session (via LightDM for example).
I don't encounter this "problem" with LXDE or Openbox sessions for example. With them, I can connect remotely through xrdp and locally at the same time.
Is there anything that I can do to fix that?
Thanks! :)

**Comment by matt335672** [comment, marked as answer]
In general, if you're running systemd, I'm afraid the answer is 'no'. This is mentioned [in the FAQ](https://github.com/neutrinolabs/xrdp/wiki/Tips-and-FAQ#why-cant-i-log-the-same-user-on-on-the-graphical-console-and-over-xrdp-at-the-same-time).

There are some provisos:-
- As you've found, some simple configurations that don't make much use of D-Bus may work.
- There is more info in #2491, inclusing a link to an alpha-quality tool which may work for some use-cases.

**Comment by pharaonic-faery** [comment]
Thank you for answering me. Excuse me, I thought the issue had to do with Xfce specifically

BTW, you can actually log in locally and over xrdp with the same user at the same time, but with two different graphical shells. For example, if you've logged in locally, you can execute `sudo update-alternatives --config x-session-manager` to change the default graphical shell, then log in over xrdp.
More simply, just log in locally choosing a graphical session other than the default one if your display manager provides a "session" button, then log in over xrdp.
And other workarounds...

But it's true that it results in unconvenient behavior. For example, if you launch a graphical app from one session, minimize it in the notification area, unhide it from the other session, then it will pops up in the session in which it had been launched from.

**Reply by matt335672** [reply]
There are lots of problems you will encounter, even if you get two sessions running. Applications and notifications popping up on the wrong desktop are commonplace I'm afraid.

You might find a workaround that's OK for you, and then a subsequent software update will break everything again.

If you are able to use a separate user for xrdp sessions (`pi-xrdp` maybe?) you should be fine. That's our current recommendation if you don't want to spend a lot of time on this.
